### DISCUSSION TITLE
Watchdog timer for ATTiny1614 does not work for 4S and 8S

### DISCUSSION THREAD

**Discussion by ghost** [SpenceKonde/megaTinyCore#1073] [Library discussion]
wdt_enable() works for all timeouts from WDTO_15MS to WDTO_2S but if WDTO_4S or WDTO_8S is used the Arduino IDE 2.3.2 produces the following error messages:

In file included from E:\Users\Steven\Documents\GitHub\Arduino_2\ATTiny1614\Blink_LED_on_PA2\Blink_LED_on_PA2.ino:3:0:
E:\Users\Steven\Documents\GitHub\Arduino_2\ATTiny1614\Blink_LED_on_PA2\Blink_LED_on_PA2.ino: In function 'void setup()':
E:\Users\Steven\Documents\GitHub\Arduino_2\ATTiny1614\Blink_LED_on_PA2\Blink_LED_on_PA2.ino:49:14: error: 'WDTO_4S' was not declared in this scope
   wdt_enable(WDTO_4S);
              ^
E:\Users\Steven\Documents\GitHub\Arduino_2\ATTiny1614\Blink_LED_on_PA2\Blink_LED_on_PA2.ino:49:14: note: suggested alternative: 'WDTO_2S'

exit status 1

Compilation error: 'WDTO_4S' was not declared in this scope

I can set the watchdog timer to 4S or 8S using the following code:
const uint8_t reset_timeout = 0xA; //or 0xB
wdt_enable(reset_timeout);

**Comment by pcfreak1201** [comment]
Hi!
When I read https://github.com/SpenceKonde/DxCore/blob/master/megaavr/extras/Ref_Reset.md correctly, you have to use WDT_PERIOD_8KCLK_gc for 8s and WDT_PERIOD_4KCLK_gc for 4s

**Reply by pcfreak1201** [reply]
(DxCore and megaTinyCore are quite similar and it is defined here: https://github.com/SpenceKonde/megaTinyCore/blob/7e836cd37f56b706c6a2f5c553ef1a6e97bf8ef9/megaavr/extras/ioheaders/iotn1614.h)

**Reply by BenthamSoftwareServices** [reply]
[Using the Watchdog timer.pdf](https://github.com/SpenceKonde/megaTinyCore/files/14807306/Using.the.Watchdog.timer.pdf)
There seems to be multiple ways of using the WDT, I've tried to document them in the attached PDF.  I think the 4 and 8 second period macro/function in wdt.h do not work for the ATTiny1614 MCU that I am using, but manually coding and using macros in iotn1614.h does work as you said - what do you think?
